### DISCUSSION TITLE
`request` does not unsusbcribe on timeout when `noMux=true`

### DISCUSSION THREAD

**Discussion by davidgf** [nats-io/nats.node#630] [Q&A] [answered]
I'm using the JS `nats@2.26.0` library, along with the `nats:2.10.14` docker image. I'm noticing a behaviour I didn't anticipate with the `request-reply` pattern. Considering the following snippet:

```
const response = await natsConnection
  .request(`requests.${msgId}`, jsonCodec.encode(message), {
    timeout,
    noMux: true,
    reply: `responses.${msgId}`,
  })
  .catch((err) => {
    console.error(err, 'Error sending NATS message');
    throw err.code === ErrorCode.Timeout ? new Error('Timeout') : new Error('Another kind of error');
  });
```

When there's a timeout error, the cli command `nats server report connections` reports a continuously increasing number of subscriptions. When the `.request()` call succeeds, this doesn't happen and the number of subscriptions remains stable. This leads me to think that the underlying subscription created by the `.request()` method on the `reply` subject is not closed automatically on timeout. If my analysis is correct and I'm not misinterpreting the reports, I wanted to know if this is the expected behaviour. 
If so, I would also like to know how the subscription should be closed.

**Comment by aricart** [comment, marked as answer]
This is fixed by https://github.com/nats-io/nats.deno/pull/715
